service/contractsv1/api.go

```diff
@@ -0,0 +1,194 @@
+package contractsv1
+
+import "fmt"
+
+func depthAllowedValues(val string) (ok bool) {
+	values := []string{"TOP", "ALL"}
+	for i := range values {
+		if ok = values[i] == val; ok {
+			return
+		}
+	}
+	return
+}
+
+// ListContracts gets the list of contracts that a user has access to.
+// 'depth' returns a specific set of contracts.
+// Select TOP to return only parent contracts or ALL to return both parent and child contracts.
+func (c *Contractsv1) ListContracts(depth string) (*OutputContractIDs, error) {
+	query := map[string]string{}
+	if depth != "" {
+		if !depthAllowedValues(depth) {
+			return nil, fmt.Errorf("Unsupported argument 'depth' value. Use 'TOP' or 'ALL' value. You provided %s", depth)
+		}
+
+		query["depth"] = depth
+	}
+
+	apiURI := fmt.Sprintf("%s/contracts/identifiers", basePath)
+
+	// Create and execute request
+	resp, err := c.Client.Rclient.R().
+		SetResult(OutputContractIDs{}).
+		SetError(ContractsErrorv1{}).
+		SetQueryParams(query).
+		Get(apiURI)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.IsError() {
+		e := resp.Error().(*ContractsErrorv1)
+
+		return nil, e
+	}
+
+	return resp.Result().(*OutputContractIDs), nil
+}
+
+// ListProductsPerContract gets the IDs and names of the products associated with a contract for the time frame selected.
+// From - Ex: 2015-01-31. The start date, in UTC, to use when looking for products associated with a contract.
+//        The search always begins at midnight (0:00) UTC of the specified date.
+//        The default start date is 30 days prior to the current date.
+//        For current contracts, you can select a date within the past 15 months of the current date.
+//        For expired contracts, you are limited to a date range of 30 days within the 15 month window.
+// To - Ex: 2016-03-31. The end date, in UTC, to use when looking for products associated with a contract.
+//      The search always ends at 23:59:59 UTC of the specified date. The default end date is the current date.
+func (c *Contractsv1) ListProductsPerContract(contractID, from, to string) (*OutputContracts, error) {
```

Comment:
Use maybe timespan ?

Reply:
You can, but do not want to add complexity here, can a feature request later to validate that given string is in right format, but that will be done by Akamai as well